openwisp_monitoring/monitoring/base/models.py

```diff
@@ -171,30 +170,11 @@ def write(self, value, time=None, database=None, check=True, extra_values=None):
             return
         self.check_threshold(value, time)
 
-    def read(self, since=None, limit=1, order=None, extra_fields=None):
+    def read(self, **kwargs):
         """ reads timeseries data """
-        fields = self.field_name
-        if extra_fields and extra_fields != '*':
-            fields = ', '.join([fields] + extra_fields)
-        elif extra_fields == '*':
-            fields = '*'
-        q = 'SELECT {fields} FROM {key}'.format(fields=fields, key=self.key)
-        tags = self.tags
-        conditions = []
-        if since:
-            conditions.append("time >= {0}".format(since))
-        if tags:
-            conditions.append(
-                ' AND '.join(["{0} = '{1}'".format(*tag) for tag in tags.items()])
-            )
-        if conditions:
-            conditions = 'WHERE %s' % ' AND '.join(conditions)
-            q = '{0} {1}'.format(q, conditions)
-        if order:
-            q = '{0} ORDER BY {1}'.format(q, order)
-        if limit:
-            q = '{0} LIMIT {1}'.format(q, limit)
-        return list(query(q, epoch='s').get_points())
+        return TimeseriesDB().read(
```

Comment:
I don't like that `TimeseriesDB` is being instantiated each time.

Either it needs to be instantiated once and used as a singleton or we should switch to using class methods only.

I think the best path would be to instantiate it only once, and perform some basic checks during instantiation (eg: does the DB exist and can we connect to it? If not fail).

Reply:
+1 to instantiating it only once